### PR TITLE
Refactor util and add util.warnings

### PR DESCRIPTION
**Pull request by Hoxbro**
At first, this PR was meant only to give a visible warning to users when running `pn.Pane`.

I wanted to add a `find_stack_level` and the logical place to have it was in `util.py`, but I felt like it would pollute the file. So I moved it to `util/__init__.py`  and split the checks into `util/checks.py`, and then imported them back into the `__ini__.py`. This should make it backward-compatible. And because I already was in these files, I made some small refactoring mostly related to type hints. 

Finally, I created a `util/warnings.py` and created a `deprecated` function, which needs a version when the function is removed from Panel. I have added an error for when this version is exceeded by Panel itself.  

Some questions:
- I would like to change all the check functions to start with `is_XXX`. But for now, I haven't done it. 
- Should I add a `deprecated` to other things, here I'm mainly thinking about `pn.pane.VTK`.

### CODE REVIEW

File: panel/tests/test_param.py [hunk 1]
```diff
@@ -25,6 +25,10 @@
 )
 
 
+def Pane(obj, **kwargs):
+    return PaneBase.get_pane_type(obj, **kwargs)(obj, **kwargs)
```
Comment (philippjfr) [inline panel/tests/test_param.py:29]: What's up with this? Not easy to replace with `pn.panel`?

Comment (Hoxbro) [reply panel/tests/test_param.py:29]: It is for functions that expect a `pn.Param` output.  `pn.panel` give a `Column` output. 

I could also change the type to assert with to `Column` if this is more desirable for you. 

``` python
def test_instantiate_from_class():

    class Test(param.Parameterized):

        a = param.Number()

    assert isinstance(Pane(Test), Param)
```

Comment (philippjfr) [reply panel/tests/test_param.py:29]: Gotcha

**Review by philippjfr — APPROVED** [review]
Thanks @Hoxbro! Looks good to me.

**Pull request merged**